src/nupic/regions/VectorFileSensor.cpp

```diff
@@ -167,14 +167,14 @@ void VectorFileSensor::compute()
 
   if (hasCategoryOut_)
   {
-    Real * categoryOut = (Real32 *) categoryOut_.getBuffer();
+    Real * categoryOut = reinterpret_cast<Real *>(categoryOut_.getBuffer());
```

Comment:
`categoryOut_` is explicitly initialized to 32 bits so I think you should change the LHS to be Real32 and leave the RHS alone. Same for below with `resetOut_`.